crates/router/src/connector/fiserv/transformers.rs

```diff
@@ -287,56 +419,150 @@ impl TryFrom<&types::PaymentsCaptureRouterData> for FiservCaptureRequest {
+        let auth: FiservAuthType = FiservAuthType::try_from(&item.connector_auth_type)?;
+        Ok(Self {
+            merchant_details: MerchantDetails {
+                merchant_id: auth.merchant_account,
+                terminal_id: None,
+            },
+            reference_transaction_details: ReferenceTransactionDetails {
+                reference_transaction_id: item
+                    .request
+                    .connector_refund_id
+                    .clone()
+                    .ok_or(errors::ConnectorError::RequestEncodingFailed)?,
+            },
+        })
+    }
 }
 
-impl From<RefundStatus> for enums::RefundStatus {
-    fn from(item: RefundStatus) -> Self {
-        match item {
-            RefundStatus::Succeeded => Self::Success,
-            RefundStatus::Failed => Self::Failure,
-            RefundStatus::Processing => Self::Pending,
-        }
+#[derive(Default, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FiservRefundRequest {
+    amount: Amount,
+    merchant_details: MerchantDetails,
+    reference_transaction_details: ReferenceTransactionDetails,
+}
+
+impl<F> TryFrom<&types::RefundsRouterData<F>> for FiservRefundRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
+        let auth: FiservAuthType = FiservAuthType::try_from(&item.connector_auth_type)?;
+        let metadata = item
+            .connector_meta_data
+            .clone()
+            .ok_or(errors::ConnectorError::RequestEncodingFailed)?;
+        let session: SessionObject = metadata
+            .parse_value("SessionObject")
+            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+        Ok(Self {
+            amount: Amount {
+                total: utils::to_currency_base_unit(
+                    item.request.refund_amount,
+                    item.request.currency,
+                )?,
+                currency: item.request.currency.to_string(),
+            },
+            merchant_details: MerchantDetails {
+                merchant_id: auth.merchant_account,
+                terminal_id: Some(session.terminal_id),
+            },
+            reference_transaction_details: ReferenceTransactionDetails {
+                reference_transaction_id: item.request.connector_transaction_id.to_string(),
+            },
+        })
     }
 }
 
-#[derive(Default, Debug, Clone, Serialize, Deserialize)]
-pub struct RefundResponse {}
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct RefundResponse {
+    gateway_response: GatewayResponse,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+#[serde(transparent)]
+pub struct RefundSyncResponse {
+    sync_responses: Vec<FiservPaymentsResponse>,
```

Comment:
If both Psync and RSync response are same keep one struct as "SyncResponse" and implement try from `ResponseRouterData`  and `RefundsResponseRouterData `